OVH DNS Provider (#143)

masterzen opened the pull request:
This adds the OVH provider along with its documentation:

- [x] DNS Provider
- [x] Registrar Provider
- [x] documentation

A few comments:
* the provider unfortunately can't use `CanUsePTR` because OVH supports only setting PTR with a target in the arpa zone. This makes the integration tests 31 and 32 fail horribly (can we make those use the arpa zone?)
* OVH now supports setting apex NS records, so dual providers scenario can be accomplished 
* OVH doesn't support CAA (I'm going to ask them when they plan to support those).

masterzen commented:
I've rebased to the latest master.

tlimoncelli commented:
Sorry for the delay.

Could you rebase  and re-run the integration tests?  I'd be glad to merge this.

masterzen commented:
@tlimoncelli I'll do that this week-end. Thanks!

masterzen commented:
@tlimoncelli,

I've rebased and made sure documentation is generated (I also implemented the `docNotes`). 

I've also enabled support for TLSA, but that required a modification of the integration test because OVH doesn't allow TLSA records for a sha256 matching type with a checksum that doesn't have exactly 64 characters (this makes sense). I couldn't test the other providers with this integration test change, so I hope I haven't introduced any test regressions.
If that proves that it is an issue, let me know and I'll revert that part of the change and exclude those tests for the OVH provider.

Thanks!

tlimoncelli commented on providers/ovh/ovhProvider.go:
```diff
@@ -0,0 +1,199 @@
+		}
+		actual = append(actual, rec)
+	}
+
```
Insert this (all providers now must do this; though if you were passing the integration tests it might be a no-op):

```
  // Normalize
  models.Downcase(actual)
```

masterzen replied:
Oh, I missed this change, I'll do that.

tlimoncelli commented on integrationTest/providers.json:
```diff
@@ -66,5 +66,13 @@
     "domain": "$VULTR_DOMAIN"
+  },
+  "OVH": {
+    "COMMENT": "47-48 OVH doesn't support TLSA records with matching type 0 (full certificate)",
```
Is there a matchingtype other than 0 that the tests could use so that the tests can run? (I don't know much about TLSA)

masterzen replied:
Yes, matching type 2 is for sha512, but since they check the length of the checksum we'd have to have a string of 128 hex characters.
Do you want me to modify the integration test to check changing matching type to 2 instead of 0 ?

tlimoncelli replied:
Sure!  The extra 128 bytes won't fill my hard disk :-) . (but seriously... I'd rather have working tests than not)

masterzen commented:
@tlimoncelli I've pushed the fixes you asked for. Let me know if there's anything I should do.

tlimoncelli commented:
Thank you for your contribution!
